### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by milosvukadinovic**

### CODE REVIEW

File: scripts/app.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+const bookArray = []; // global array
+
+// MVP part
+/**
+ * This is the booj object
+ * @param {string} title The title of the book
```
Comment (rvvergara) [inline scripts/app.js]: What is the purpose of lines 6-9 (usage of `@param`)?

File: scripts/app.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  author.value = '';
+  numPages.value = '';
+  // re-rendering the table of data
+  render();
+}
+/**
+* this function is the handler of the checkbox  and it will toggle ita
+* @param{event} event is the event
+*/
+function toggleIsRead(event) {
```
Comment (rvvergara) [inline scripts/app.js]: This function should be in the `Book` 's prototype and should not be this complex. This function `toggleIsRead` should simply change the value of `isRead` from `true` to `false` or vice-versa. Let's transfer this into the `Book` 's prototype.

File: scripts/app.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  render();
+}
+// Listeners
+document.getElementById('submitBookBtn').addEventListener('click', createNewBook);
+
+/**
+* The handler of the button 'Submit book'
+@param{e} e is the event
+*/
+function createNewBook(e) {
```
Comment (rvvergara) [inline scripts/app.js]: Let's have a separate function of adding new book into the library that has nothing to do with the DOM. The function that we have now mixes DOM and library related logic.

```javascript
// this should be how your addBookToLibrary is structured:
function addBookToLibrary(libaryArr, title, author, pages, isRead) = {
  const book = new Book(title, author, pages, isRead);
  libraryArr.push(book);
  // optionally returns the new book
  return book;
}

// then we can call it like this:
const newBook = addBookToLibrary(bookArray, "Harry Potter", "JK Rowling", 250, true)
```

What this does is simply create a new book and push that new book into your app's library.

**Review by rvvergara — CHANGES_REQUESTED** [review]
Hi @azdrenymeri and @milosvukadinovic,

Congratulations and welcome to the JavaScript world! Great start. We just need to work out some things:

### Let us use a .gitignore file

Your package is using dependencies and whenever a user installs those dependencies, it's not good to have them included in the repository. We SHOULD have a .gitignore file that contains the `node_modules/` line.

Here's our Project Checklist, most of which you guys have satisfied but a few that are in fact important we still need to accomplish:
#### Application
- [x] It has an array to store the instances of `Book`

##### Book
- [x] It has a constructor

##### `addBookToLibrary` function
- [x] The application has a `addBookToLibrary` function
- [x] It can take user's input
- [x] It stores the new Book objects into an array

##### `New Book` button
- [x] It has a `New Book` button
- [x] It brings up a form
- [x] The form has an `author` field
- [x] The form has a `title` field
- [x] The form has a `number of pages` field
- [x] The form allows to mark the `Book` status as `read`

##### Display `Book`
- [x] It has a `render` function
- [x] It displays on the page every book contained in the array
- [x] Each `Book` displayed has a `button` to `remove` the `Book` from the `Library` array
- [x] Each `Book` displayed has a `button` to `toggle` the `Book` read status

### Crucial things to make project work

- [x] The project has a descriptive README.md
- [x] Verify if the project should be done collaboratively => **Based on your repo stats, @milosvukadinovic only contributed 2 commits out of all the 15 total. What happened here?** 
- [x] Keep the application logic separated from DOM manipulation tasks -> Check my annotations on app.js on this
- [x] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript)

File: scripts/app.js (hunk 1 above)
Comment (milosvukadinovic) [reply scripts/app.js]: Azdren was showing me how eslint rules work,

File: scripts/app.js (hunk 2 above)
Comment (milosvukadinovic) [reply scripts/app.js]: Resolved

File: scripts/app.js (hunk 1 above)
Comment (azdrenymeri) [reply scripts/app.js]: Lines from 6 to 9 explain the parameters of the book object this js-doc comment is enforced by eslint we can add a rule into eslint config JSON file to ignore it

File: scripts/app.js (hunk 3 above)
Comment (milosvukadinovic) [reply scripts/app.js]: Fixed

**Review by rvvergara — CHANGES_REQUESTED** [review]
### Node modules are still contaminating the repository.

Let's clean up our repo by removing the node_modules folder from the `master` branch.

Please remember when you're building a new JavaScript project to always always include `node_modules` in your `.gitignore` file. 

### Let's remove all commented code.

### Book.prototype.toggleIsRead contaminated with DOM related code

Let's not do this. This function should be simply toggling a book object's `isToggleRead` attribute. We don't need any DOM element (in this case why did we need `index`) to do this.

**Comment by azdrenymeri** [comment]
> ### Node modules are still contaminating the repository.
> Let's clean up our repo by removing the node_modules folder from the `master` branch.
 ### Done

> Please remember when you're building a new JavaScript project to always always include `node_modules` in your `.gitignore` file.
### We will do it on future projects thanks

> ### Let's remove all commented code.
#### I removed all jsdoc comments (but still there are some small section commented just to understand the code in future)

> ### Book.prototype.toggleIsRead contaminated with DOM related code
> Let's not do this. This function should be simply toggling a book object's `isToggleRead` attribute. We don't need any DOM element (in this case why did we need `index`) to do this.
### Fixed
#### I added index as a parameter to the method `toggleIsRead` and when calling on line [72](https://github.com/azdrenymeri/library/blob/c16ec329fe0de7ecdfa692adb0aa245d2f424e8d/scripts/app.js#L72)  it will pass the index

**Review by rvvergara — APPROVED** [review]
Hi @azdrenymeri and @milosvukadinovic,

There's just one but very important change you need to make (see below). Though, I won't hold you much longer on this project because of that. I will stamp this with my approval. I just recommend to you guys to review the concepts you need to learn for this project. Based on the changes we talked about there are some concepts that may need more clarification. For instance the role of the `prototype` object and the `this` keyword. Another is how to separate concerns from each other (one of the primary mantras of OOP). Having said this you did well in most parts of the project. You wrote your constructors very well and you have a clean and very understandable User Interface.

### Book.prototype.toggleIsRead

I was thinking of a simpler and a more proper approach. Currently this method is 1) Using a global object (`bookArray`) which isn't a good way to structure your `Book`, 2) It doesn't need any index as we have access to the `this` keyword:

```javascript
Book.prototype.toggleIsRead = function(){
  this.isRead = !this.isRead;
};
```

Apart from this I don't think there's anything more I'd add so:

# :+1: 

You may submit your project completion form and merge this into `master`.

**Pull request closed without merging**